src/types/custom.ts

```diff
@@ -0,0 +1,20 @@
+import { Runtype, Rt, Static, create, validationError } from '../runtype'
+import { String } from './string'
+
+export interface Custom<A extends Rt, T extends string, K> extends Runtype<Static<A>> {
+  tag: T,
```

Comment:
So that you can write functions against a `Reflect` and know that all the possible alternatives (what the values of `tag` might be) have been handled. For example, look at the definition of [`show`](https://github.com/typeetfunc/runtypes/blob/feature/custom-type/src/show.ts). It defines a switch statement across all possible run types and it doesn't need a `default` case because it knows it's handled everything. If you deleted any of those cases or if you added another alternative to `Reflect` it would be a static type error (assuming your compiler settings are appropriately strict), which is nice because it alerts you to potential bugs.